uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/OrchestratorZoneControllerIntegrationTests.java

```diff
@@ -716,23 +734,43 @@ private ResponseEntity<Void> createIdentityZone(String zoneId) {
         OAuth2RestTemplate restTemplate = new OAuth2RestTemplate(identityZoneClientCredentials, new DefaultOAuth2ClientContext());
 
         //Calling the API to call Identity Zone
-
-        String requestBody = "{\"id\":\"" + zoneId + "\", \"subdomain\":\"" + zoneId + "\", \"name\":\"testImportExistingZone() " + zoneId + "\"}";
+        String requestBody = "{\"id\":\"" + zoneId + "\", \"subdomain\":\"" + zoneName + "\", \"name\":\" " + zoneName + "\"}";
 
         HttpHeaders headers = new HttpHeaders();
         headers.add("Accept", MediaType.APPLICATION_JSON_VALUE);
         headers.add("Content-Type", MediaType.APPLICATION_JSON_VALUE);
 
-        ResponseEntity<Void> response = restTemplate.exchange(
-                serverRunning.getUrl("/identity-zones"),
-                HttpMethod.POST,
-                new HttpEntity<>(requestBody, headers),
-                new ParameterizedTypeReference<Void>() {
-                });
+        ResponseEntity<Void> response = restTemplate.exchange(serverRunning.getUrl("/identity-zones"), HttpMethod.POST, new HttpEntity<>(requestBody, headers), new ParameterizedTypeReference<Void>() {
+        });
 
         return response;
     }
 
+    private void createClient(String zoneId, final String authorities, final String clientId, final String clientSecret, final String grantTypes, final String resourceIds, final String scopes) {
+        String accessToken = IntegrationTestUtils.getClientCredentialsToken(serverRunning, testAccounts.getAdminClientId(), testAccounts.getAdminClientSecret());
+        BaseClientDetails client = new BaseClientDetails(ADMIN_CLIENT_NAME, resourceIds, scopes, grantTypes, authorities);
+        client.setClientSecret(ADMIN_CLIENT_SECRET);
+
+        URI currentURI = URI.create(serverRunning.getUrl(OAUTH_CLIENT_URI));
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+        headers.add("X-Identity-Zone-Id", zoneId);
+        headers.add(AUTHORIZATION, "Bearer " + accessToken);
+
+        HttpEntity<BaseClientDetails> requestEntity = new HttpEntity<>(client, headers);
+
+        ResponseEntity<BaseClientDetails> createClientResponse = new RestTemplate().postForEntity(currentURI, requestEntity, BaseClientDetails.class);
+
+        Assert.assertEquals(HttpStatus.CREATED, createClientResponse.getStatusCode());
+    }
+
+    private void validateToken(String zoneUrl, String zoneId) {
+
+        OAuth2RestTemplate zoneAdminClient = (OAuth2RestTemplate) IntegrationTestUtils.getClientCredentialsTemplate(IntegrationTestUtils.getClientCredentialsResource(zoneUrl, new String[0], ADMIN_CLIENT_NAME, ADMIN_CLIENT_SECRET));
+        validateZoneAdminClient(zoneUrl, zoneId, zoneAdminClient);
```

Comment:
Validating zone admin client might be an overkill... enough just to get a token and validate that it's not null

`
String accessToken = IntegrationTestUtils.getClientCredentialsToken(zoneUrl, ADMIN_CLIENT_NAME, ADMIN_CLIENT_SECRET);
`
`
assertNotNull(accessToken);
`

Reply:
Changed as suggested. Instead of validating the client , just validating if token is being generated